### DISCUSSION TITLE
PLD Royal Authority > Atonement chain not working

### DISCUSSION THREAD

**Discussion by deriatt** [daemitus/XIVComboPlugin#351] [1 - Tanks]
When I have combo #128 enabled (Royal Authority Atonement Feature), Royal Authority changes to atonement only for a second then falls back to Fast Blade. If I have the atonement chain on another button and use it, my Royal Authority button correctly changes to each part of the combo but only for a second before switching back to fast blade.

This behavior is evident regardless of whether the conflicting skill #131 is disabled. It doesn't appear that 131 does anything.

**Comment by MKhayle** [comment]
This is an issue I am aware and which is being worked on! 
It's actually working, but for many reasons, it's part of a bigger rework because it will only "work" if the buff would get replaced by another one (so during a second combo chain) which is an optimal behaviour but unintuitive one

**Reply by deriatt** [reply]
Thanks for the quick response! I hadn't noticed that it was working fine on the 2nd loop, I'd just been testing the combos in a vacuum and it seemed wrong but this behavior does make perfect sense, thanks for the clarification.

**Reply by MKhayle** [reply]
Yes, I am designing a new way to separate intuitive and optimal behaviours for both beginners and more "hardcore" players. Sorry for the current delay!

**Reply by kaedys** [reply]
To clarify, the main reason it works the way that it does right now is because it is generally optimal to delay using Atonement until it would be replaced by another usage of Royal Authority, rather than using it immediately.  This is because delaying usage like that means that at any time, you can guarantee that your next 3 GCDs are all amongst your highest potency attacks, which is vital for Fight or Flight optimization.

**Warning: Paladin theorycrafting details**

Paladin optimization at the moment is mostly focused on packing as much potency as possible into the 3 "spare" GCDs during Fight or Flight (FoF lasting 8 GCDs, and 5 of those consumed by the 4-part Confiteor combo and Goring Blade).  The potency order is currently Sepulchre (480p) >  Holy Spirit w/ Divine Might (470p) > Supplication (460p) > Atonement (440p) = Royal Authority (440p) >> Riot Blade (330p) >> Fast Blade (220p).  As you can see, the first 5 options are all pretty close in potency.  The key element is just avoiding using the much lower potency Riot Blade or Fast Blade during the FoF combo period.  Delaying using Atonement and Holy Spirit until Royal Authority would be the next cast _guarantees_ that at any given time, your next 3 GCDs can _always_ be composed entirely of just those first 5 high-potency attacks.  

The absolute highest potency sequence is Supplication + Sepulchre + Holy Spirit, which is 1410p, with the other possible sequences sitting somewhere between that and 1350p.  Really, as long as you avoid using Riot Blade or Fast Blade, which you always can if you delay Atonement and Divine Might usage until it would be overwritten by another Royal Authority, you're getting most of the benefit already.  With the priority that the mod currently uses, the _minimum_ potency sequence you can end up using for those 3 GCDs is Royal Authority + Holy Spirit + Atonement, which is only a 1350p and thus at most a 60 potency loss over the "optimal".

If you think about the standard cycle used by this mod currently:

`Fast Blade -> Riot Blade -> Atonement -> Supplication -> Sepulchure -> Holy Spirit -> Royal Authority`

If your next normal GCD would by the first ability below, but you activate FoF, this is the sequence and potency you'll get using this mod's logic:

* Fast Blade => `Atonement + Supplication + Sepulchre (1380p)`
* Riot Blade => `Atonement + Supplication + Sepulchre (1380p)`
* Atonement => `Atonement + Supplication + Sepulchre (1380p)`
* Supplication => `Supplication + Sepulchre + Holy Spirit (1410p)`
* Sepulchre => `Sepulchre + Holy Spirit + Royal Authority (1390p)`
* Holy Spirit => `Holy Spirit + Royal Authority + Holy Spirit (1380p)`
* Royal Authority => `Royal Authority + Holy Spirit + Atonement (1350p)`

So every single result is within 60p of the optimal, and the average result is ~1381.5 potency.  Without the delay of Atonement and Holy Spirit, where they are used immediately, the resulting normal sequence is:

`Fast Blade -> Riot Blade ->  Royal Authority -> Atonement -> Supplication -> Sepulchure -> Holy Spirit`

And the resulting FoF windows are:

* Fast Blade => `Fast Blade + Riot Blade + Royal Authority (990p)`
* Riot Blade => `Riot Blade + Royal Authority + Holy Spirit (1240p)`
* Royal Authority => `Royal Authority + Holy Spirit + Atonement (1350p)`
* Atonement => `Atonement + Supplication + Sepulchre (1380p)`
* Supplication => `Supplication + Sepulchre + Holy Spirit (1410p)`
* Sepulchre => `Sepulchre + Holy Spirit + Fast Blade (1170p)`
* Holy Spirit => `Holy Spirit + Fast Blade + Riot Blade (1020p)`

Average result is now only ~1223 potency, a staggering ~158.5 potency loss, on average, per minute, and a maximum loss of 390 potency.  Ouchy =(
